### DISCUSSION TITLE
SSH authentication

### DISCUSSION THREAD

**Discussion by Vince1024** [Natizyskunk/vscode-sftp#456] [Q&A] [answered]
Hi all,

I love this extension and use it with many servers.

With some of them (old ones) I have a problem to connect to and getting this error message :

"Unable to negotiate with xxx.xxx.xxx.xxx port 22: no matching host key type found. Their offer: ssh-dss"

Can you help me to get the proper config to connect these servers ?

I try adding some more keys in my config like **"algorithms": { ... }** but nothing works for me.

Current config:
```
{
      "name": "Srv-name",
      "context": "My_ctx",
      "protocol": "sftp",
      "host": "myServer",
      "port": 22,
      "username": "My UserName",
      "password": "MypA@s$",
      "remotePath": "/home/users"
}
```

Thanks for your help.

**Comment by Natizyskunk** [comment, marked as answer]
@Vince1024,
I recommend using an SSH private Key file instead. You can then add the `PubkeyAcceptedKeyTypes` instruction at the top of your local `.ssh/config` file to make it work. You can also define per Host specific algorithms to use for the connection.

`sftp.json` file :
```
{
    [...]
    "protocol": "sftp",
    "port": 22,
    "secure": true,
    "sshConfigPath": "~/.ssh/config",
    "privateKeyPath": "~/.ssh/mywebsite.com/id_rsa"
}
```

`.ssh/config` file :
```
PubkeyAcceptedKeyTypes=+ssh-dss,ssh-rsa
[...]
Host mywebsite.com 192.100.100.1
    HostName mywebsite.com
    User username
    PreferredAuthentications publickey
    IdentityFile ~/.ssh/mywebsite.com/id_rsa
 ```
